src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilderHelper.cs

```diff
@@ -358,6 +358,11 @@ private void ProcessLog(string currentLog)
          * */
         private void BuildSolution(IAnalyzerManager manager)
         {
+            if (manager.Projects.Count <= 0)
+            {
+                Logger.LogError($"Solution {manager.SolutionFilePath} does not have any projects");
+                return;
```

Comment:
Will it cause any issues inside the callers of this if the caller thinks something is going to be built but nothing gets built? Can we add a test to make sure?

Reply:
Follow-up question: Should we even try building the solution if it has zero projects? I.e. Should we check for the existence of projects before we call `BuildSolution(...)`?

Reply:
@Beau-Gosse-dev added test, and verified we handling it well and return a result object with no results.

@jonlouie I think we're essentially doing this, it's just the location of the check is inside the method instead of outside?